contracts/adapters/Hyperlane/HyperlaneSenderAdapter.sol

```diff
@@ -55,26 +65,39 @@ contract HyperlaneSenderAdapter is IBridgeSenderAdapter, BaseSenderAdapter, Owna
      * @notice HyperlaneSenderAdapter constructor.
      * @param _mailbox Address of the Hyperlane `Mailbox` contract.
      */
-    constructor(address _mailbox) {
+    constructor(address _mailbox, address _igp) {
         if (_mailbox == address(0)) {
             revert Errors.InvalidMailboxZeroAddress();
         }
         mailbox = IMailbox(_mailbox);
+        _setIgp(_igp);
     }
 
     /// @inheritdoc IBridgeSenderAdapter
     /// @dev we narrow mutability (from view to pure) to remove compiler warnings.
     /// @dev unused parameters are added as comments for legibility.
     function getMessageFee(
-        uint256 /* toChainId*/,
+        uint256 toChainId,
         address /* to*/,
         bytes calldata /* data*/
-    ) external pure override returns (uint256) {
-        // Hyperlane fee can't be calculated based on these inputs, we return 0 and leave fee payment to another transaction/agent
+    ) external view override returns (uint256) {
+        uint32 dstDomainId = _getDestinationDomain(toChainId);
+        // An IGP and dstDomainId has to be properly set for gas quotation to work
         // See https://docs.hyperlane.xyz/docs/build-with-hyperlane/guides/paying-for-interchain-gas
+        if (address(igp) != address(0) && dstDomainId != 0) {
+            // destination gasAmount is hardcoded to 500k similar to Wormhole implementation
+            return igp.quoteGasPayment(dstDomainId, 500000);
+        }
+
+        // Default to zero, MultiMessageSender.estimateTotalMessageFee doesn't expect this function to revert
```

Comment:
If this is the case, should there by a try-catch around `quoteGasPayment?

Reply:
good catch, there should be, adding